src/tracing/mux.rs

```diff
@@ -0,0 +1,388 @@
+    pub fn try_from_config(config: MuxConfig) -> Result<MuxInspector, Error> {
+        let inspectors = config
+            .tracers
+            .into_iter()
+            .map(|(tracer_type, tracer_config)| {
+                MultiInspector::try_from_config(tracer_type, tracer_config.clone())
+            })
+            .collect::<Result<Vec<_>, _>>()?;
+
+        Ok(MuxInspector { inspectors })
+    }
+
+    /// Try converting this [MuxInspector] into a [MuxFrame].
+    pub fn try_into_mux_frame<DB: DatabaseRef>(
+        self,
+        result: &ResultAndState,
+        db: &DB,
+    ) -> Result<MuxFrame, DB::Error> {
+        let mut frame = HashMap::with_capacity(self.inspectors.len());
+        for (tracer_type, inspector) in self.inspectors {
+            let trace = match inspector {
+                MultiInspector::FourByte(inspector) => FourByteFrame::from(inspector).into(),
+                MultiInspector::Call(config, inspector) => inspector
+                    .into_geth_builder()
+                    .geth_call_traces(config, result.result.gas_used())
+                    .into(),
+                MultiInspector::Prestate(config, inspector) => {
+                    inspector.into_geth_builder().geth_prestate_traces(result, config, db)?.into()
+                }
+                MultiInspector::Noop => NoopFrame::default().into(),
+                MultiInspector::Mux(inspector) => {
+                    inspector.try_into_mux_frame(result, db).map(GethTrace::MuxTracer)?
+                }
+            };
+
+            frame.insert(tracer_type, trace);
+        }
+
+        Ok(MuxFrame(frame))
+    }
+}
+
+impl<DB> Inspector<DB> for MuxInspector
+where
+    DB: Database,
+{
+    #[inline]
+    fn initialize_interp(&mut self, interp: &mut Interpreter, context: &mut EvmContext<DB>) {
+        for (_, inspector) in &mut self.inspectors {
+            inspector.initialize_interp(interp, context);
+        }
+    }
+
+    #[inline]
+    fn step(&mut self, interp: &mut Interpreter, context: &mut EvmContext<DB>) {
+        for (_, inspector) in &mut self.inspectors {
+            inspector.step(interp, context);
+        }
+    }
+
+    #[inline]
+    fn step_end(&mut self, interp: &mut Interpreter, context: &mut EvmContext<DB>) {
+        for (_, inspector) in &mut self.inspectors {
+            inspector.step_end(interp, context);
+        }
+    }
+
+    #[inline]
+    fn log(&mut self, context: &mut EvmContext<DB>, log: &Log) {
+        for (_, inspector) in &mut self.inspectors {
+            inspector.log(context, log);
+        }
+    }
+
+    #[inline]
+    fn call(
+        &mut self,
+        context: &mut EvmContext<DB>,
+        inputs: &mut CallInputs,
+    ) -> Option<CallOutcome> {
+        for (_, inspector) in &mut self.inspectors {
+            if let Some(outcome) = inspector.call(context, inputs) {
+                return Some(outcome);
+            }
```

Comment:
I believe the `TracingInspector` does not override the response so we can ignore this some check